src/components/data-exploration/heatmap/HeatmapPlot.jsx

```diff
@@ -163,20 +174,50 @@ const HeatmapPlot = (props) => {
       || !heatmapSettings.groupedTracks
       || !heatmapSettings.selectedCellSet
       || !heatmapSettings.selectedPoints
+      // If selectedGenes isn't null, then we are not at the initial load, so don't load markers
+      // We are only supposed to load marker genes to fill up the heatmap at the beginning)
+      || !_.isNil(selectedGenes)
     ) return;
 
     const { groupedTracks, selectedCellSet, selectedPoints } = heatmapSettings;
 
+    const downsampleSettings = {
+      groupedTracks,
+      selectedCellSet,
+      selectedPoints,
+      hiddenCellSets: cellSets.hidden,
+    };
+
     dispatch(loadMarkerGenes(
       experimentId,
       COMPONENT_TYPE,
-      {
-        numGenes: nMarkerGenes,
-        groupedTracks,
-        selectedCellSet,
-        selectedPoints,
-        hiddenCellSets: cellSets.hidden,
-      },
+      { numGenes: nMarkerGenes, ...downsampleSettings },
+    ));
+  }, [
+    louvainClustersResolution,
+    cellSets.accessible,
+    heatmapSettings?.groupedTracks,
+    heatmapSettings?.selectedCellSet,
+    heatmapSettings?.selectedPoints,
+    cellSets.hidden,
+    groupedCellSets,
+  ]);
+
+  useConditionalEffect(() => {
+    if (
+      !cellSets.accessible
+      || !louvainClustersResolution
+      || !heatmapSettings.groupedTracks
```

Comment:
Is it possible to group these things into one variable similar to what we have with `cellSets.accessible` so that here and in the other useConditionalEffect we just check something like `heatmapSettings.shouldLoadMarkerGenes` or something similar?

Reply:
Just a suggestion I don't know if this will be beneficial enough or not.

Reply:
These checks seems to be fairly specific to this component, so I don't know if it would be useful to have it accessible straight from a selector. If we grouped them it would need to be in this component I think. The thing is that these checks are a bit different for `loadMarkerGenes` vs `loadDownsampledGeneExpression` (because marker genes only need to be loaded once at the beginning and that's it).

A `accessible` was set up because there were these complicated checks on cell sets were being used in many different components (`accessible` checks many things, like loading or not loading but still hasn't begun its initial load, etc).

As a bit of unrelated info, `accessible` is not in redux, it is returned by the `getCellSets` selector.